locallibs/productbuild.py

```diff
@@ -0,0 +1,31 @@
+import os
+import shutil
+import subprocess
+import sys
+
+class ProductBuildException(Exception):
+    '''Error when creating pkg'''
+    pass
+
+def generate(info, createuserpkg_dir):
+    # Hide and rename created component package
+    component_pkg = os.path.dirname(info[u'destination_path']) + "/.tmp.pkg"
+    shutil.move(os.path.expanduser(info[u'destination_path']), component_pkg)
+
+    print "Overwriting {} for product creation".format(os.path.expanduser(info[u'destination_path']))
```

Comment:
This message sounds scarier than it probably needs to -- it makes it sound like we're stomping on some pre-existing file. You might want to take a look at how munki-pkg handles the optional creation of distribution-style pkgs.

Reply:
I agree that the message can be a little less drastic.
As I stated in my previous message:
> I considered leaving the original component package alongside the generated distribution package but I think that more than one output package is confusing so I chose to notify the user using a simple message.

Do you think it would be better to have two outputs, one for the distribution style package that has a `Dist-` prefix in the filename?

Reply:
No, I don't. If the user requested a distribution-style pkg, there should be only one output: the distribution-style pkg.
I just think the message doesn't really tell the user anything useful and might actually make the user worried that some pre-existing file is getting overwritten (that there is some sort of _problem_, when in reality this is all expected and normal).

Reply:
Then the most logical would be to not prompt the user at all, right?

Reply:
"Prompt" to me means to ask a question. Do you mean to not print anything at all? It looks like userpkg.generate doesn't print anything unless there's an error/exception, so it would make sense for the equivalent in productbuild.py to do the same.

Reply:
Yes, you're right of course it's print, not prompt. I pushed two commits again. Should be all good now. I also added a separate `OptionGroup` called Optional Package Options specifically for the `-d` flag because it doesn't make sense to add it to any of the other option groups and I can imagine there might be other optional package options in the future.